### ISSUE TITLE
jsr -x(a6) is displayed wrong

### ISSUE THREAD

**Issue by emoon** [emoon/capstone#18]
They are shown like

```ASM
jsr $ff3d(a6)
```

While this isn't fully incorrect as it shows the signed value as unsigned it would be nice to fix

**Comment by emoon** [comment]
Thinking about what is the best way to fix this as we may have the case again.

Would it makes sense to have something that specify if an imm is of a certain size? (like u16/s16) so the printer can use the correct format when printing it.

**Comment by emoon** [comment]
Could possible use imm_u16, imm_s16 and store which flavor to use when printing the imm. A bit unsure about this one.

**Comment by nplanel** [comment]
In objdump I did it in the printing, via signed_insn as there are multiple mnemonic in this case.
I think It's better to store the unsigned value (raw) and interpret/print it as you prefer (lot's of time signed)

**Comment by emoon** [comment]
Yeah true. I guess we need to add some more logic inside https://github.com/emoon/capstone/blob/next/arch/M68K/M68KDisassembler.c#L305 to figure out how to print things depending on the op or something

**Comment by nplanel** [comment]
Yeah, it's kind of voodoo by mnemonic

**Comment by emoon** [comment]
Yeah. It's not a shower stopper for sure but nice to have.
